subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/ReportState.kt

```diff
@@ -38,7 +38,10 @@ sealed class ReportState {
             internal fun getCurrentStepOrCreate(factory: () -> StepResult): StepResult {
                 return currentStep ?: factory().also { step ->
                     currentStep = step
-                    testCaseStepList.add(step)
+                    when {
+                        testCaseStepList.isNotEmpty() -> testCaseStepList.add(step)
```

Comment:
@Kutashov contributes a bug: 
- When the Synthetic step is before a precondition but there were no real steps It appeared in real steps and confused users
- Now the Synthetic step will be added to the preconditions If there are no real steps

Reply:
Do we have unit tests for this logic?

Reply:
Yes, I fix them too